src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectly.cs

```diff
@@ -0,0 +1,146 @@
+        {
+            foreach (INamedTypeSymbol anInterface in symbol.Interfaces)
+            {
+                if (IsKnownInterfaceInTheChain(anInterface) &&
+                    FirstTypeParameterNameIsNotTheSymbolName(symbol, anInterface))
+                {
+                    SyntaxNode? typeParameter = FindTheTypeArgumentOfTheInterfaceFromTypeDeclaration(symbol, anInterface);
+                    context.ReportDiagnostic(CreateDiagnostic(GMIRuleForInterface, typeParameter, anInterface.OriginalDefinition.ToDisplayString(
+                        SymbolDisplayFormat.MinimallyQualifiedFormat), anInterface.OriginalDefinition.TypeParameters[0].Name, symbol));
+                    break;
+                }
+            }
+
+            INamedTypeSymbol? baseType = symbol.BaseType;
+            while (baseType != null && baseType.IsGenericType)
+            {
+                foreach (INamedTypeSymbol anInterface in baseType.Interfaces)
+                {
+                    if (IsKnownInterfaceInTheChain(anInterface) &&
+                        FirstTypeParameterNameIsNotTheSymbolName(symbol, anInterface))
+                    {
+                        SyntaxNode? typeParameter = FindTheTypeArgumentOfTheInterfaceFromTypeDeclaration(symbol, symbol.BaseType);
+                        context.ReportDiagnostic(CreateDiagnostic(GMIRuleForClass, typeParameter, symbol.BaseType.OriginalDefinition.ToDisplayString(
+                            SymbolDisplayFormat.MinimallyQualifiedFormat), symbol.BaseType.OriginalDefinition.TypeParameters[0].Name, symbol));
+                        break;
+                    }
+                }
+
+                baseType = baseType.BaseType;
+            }
+
+            bool IsKnownInterfaceInTheChain(INamedTypeSymbol anInterface)
+            {
+                if (anInterface.IsGenericType)
+                {
+                    if (IsKnownInterface(anInterface, systemNS, systemNumericsNS))
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        foreach (INamedTypeSymbol parentInterface in anInterface.Interfaces)
+                        {
+                            if (IsKnownInterfaceInTheChain(parentInterface))
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        private static Diagnostic CreateDiagnostic(DiagnosticDescriptor GMIRule, SyntaxNode? parameter,
+            string genericDeclaration, string parameterName, INamedTypeSymbol symbol) =>
+                parameter is null ? symbol.CreateDiagnostic(GMIRule, genericDeclaration, parameterName, symbol.Name) :
+                    parameter.CreateDiagnostic(GMIRule, genericDeclaration, parameterName, symbol.Name);
```

Comment:
I think we can just always create the diagnostic on the symbol and avoid calculating the type parameter at all. This will be more consistent for cases where it's hard to determine the type parameter syntactically.

Reply:
We can, but I prefer to squiggle wherever the diagnostic apply, whenever it is possible